assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -7795,59 +7927,110 @@ private void setNewAssignmentParameters(RunData data, boolean validify) {
+                                        groupChoice != null ? Arrays.asList(groupChoice) : new ArrayList<>(), selectedCategories, true);
+
+                                if (!areCategoriesInGroups) {
+                                    addAlert(state, rb.getFormattedMessage("group.sitegradebook.categories.error"));
+                                }
                             }
+
+                            List<Gradebook> gbList = gradingService.getGradebookGroupInstances(siteId);
+                            List<String> existingGradebookUids = gbList.stream().map(Gradebook::getUid).collect(Collectors.toList());
+
+                            for (String gbUid : existingGradebookUids) {
+                                buildGradebookPointsMap(gbUid, siteId, assignmentRef, gradebookPointsMap, newCategoryString);
+                            }
+                        } else {
+                            buildGradebookPointsMap(siteId, siteId, assignmentRef, gradebookPointsMap, catInt.toString());
                         }
-                    }
-                    if (thisCatRef != -1) {
-                        for(CategoryDefinition thisCategoryDefinition : categoryDefinitions) {
-                            if (Objects.equals(thisCategoryDefinition.getId(), thisCatRef)) {
-                                if (thisCategoryDefinition.getDropKeepEnabled() && !thisCategoryDefinition.getEqualWeight()) {
-                                    Double thisCategoryPoints = thisCategoryDefinition.getPointsForCategory();
-                                    if (thisCategoryPoints != null) {
-                                        droppedCategoryPoints = thisCategoryPoints;
-                                    }
+                        break;
+                    case GRADEBOOK_INTEGRATION_ASSOCIATE:
+                        if (isGradebookGroupEnabled) {
+                            String assignTo = params.getString("assignTo");
+
+                            if (!assignTo.equals("individuals")) {
+                                List<String> gbItemList = Arrays.asList(gbSelector.split(","));
+
+                                boolean areItemsInGroups =
+                                    gradingService.checkMultiSelectorList(siteId,
+                                        groupChoice != null ? Arrays.asList(groupChoice) : new ArrayList<>(), gbItemList, false);
+
+                                if (!areItemsInGroups) {
+                                    addAlert(state, rb.getFormattedMessage("group.sitegradebook.items.error"));
                                 }
                             }
                         }
-                    }
+                        break;
+                    default:
+                        break;
                 }
 
+                //DISTINTO EN 25X - ahora hace el match por id en vez de nombre? pero creo que falta conversion?
```

Comment:
```suggestion
```

Reply:
There was a patch by @hornersa but the conversion wasn't done so the patch wasn't accepted, the same would apply here.